packages/react/src/Accordion/Accordion.features.stories.tsx

```diff
@@ -0,0 +1,207 @@
+        </Box>
+      </Stack>
+    )
+  },
+}
+
+const headings = [
+  "What's included in the GitHub for Startups offer?",
+  'Who is eligible to apply?',
+  'What if my startup is not eligible? Are there other resources for me?',
+  'How can my organization become a GitHub for Startups partner?',
+]
+
+export const AlwaysExactlyOnePanelOpen: StoryObj = {
+  render: () => {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const containerRef = useRef<HTMLDivElement>(null)
+
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const onClick = useCallback<NonNullable<AccordionRootProps['onClick']>>(e => {
+      // Prevent closing of an open accordion
+      if (e.currentTarget.open) {
+        e.preventDefault()
+        return
+      }
+
+      if (!containerRef.current) return
+
+      // Close all other accordions
+      const openAccordions = containerRef.current.querySelectorAll('details[open]')
+      for (const openAccordion of openAccordions) {
+        if (openAccordion !== e.currentTarget) {
+          openAccordion.removeAttribute('open')
+        }
+      }
+    }, [])
+
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const onKeyDownCapture = useCallback<NonNullable<AccordionRootProps['onKeyDownCapture']>>(e => {
+      // Prevent the escape key from closing the accordion
+      if (e.key === 'Escape') {
+        e.preventDefault()
+        e.stopPropagation()
+      }
+    }, [])
+
+    return (
+      <div ref={containerRef}>
+        {headings.map((heading, index) => (
+          <Accordion key={index} open={index === 0} onClick={onClick} onKeyDownCapture={onKeyDownCapture}>
+            <Accordion.Heading>{heading}</Accordion.Heading>
+            <Accordion.Content>
+              <p>
+                Lorem ipsum dolor, sit amet consectetur adipisicing elit. Blanditiis quidem veniam vero omnis
+                consequuntur cum quae libero dolor dicta odio in, corporis perspiciatis nesciunt facere. Eius vero culpa
+                quae itaque?
+              </p>
+            </Accordion.Content>
+          </Accordion>
+        ))}
+      </div>
+    )
+  },
+}
+
+export const ExclusiveUsingNameAttribute: StoryObj = {
+  render: () => (
+    <>
+      {headings.map((heading, index) => (
+        <Accordion key={index} name="exclusive">
+          <Accordion.Heading>{heading}</Accordion.Heading>
+          <Accordion.Content>
+            <p>
+              Lorem ipsum dolor, sit amet consectetur adipisicing elit. Blanditiis quidem veniam vero omnis consequuntur
+              cum quae libero dolor dicta odio in, corporis perspiciatis nesciunt facere. Eius vero culpa quae itaque?
+            </p>
+          </Accordion.Content>
+        </Accordion>
+      ))}
+    </>
+  ),
+}
+
+export const ExclusiveWithoutUsingNameAttribute: StoryObj = {
+  render: () => {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const containerRef = useRef<HTMLDivElement>(null)
```

Comment:
Is there another way to call this hook outside of a render method to avoid the rule of hook violation?